src/main/java/ch/njol/skript/effects/EffToggle.java

```diff
@@ -123,13 +133,23 @@ public boolean init(final Expression<?>[] vars, final int matchedPattern, final
 	
 	@Override
 	protected void execute(final Event e) {
+		Object[] values = expressions.getArray(e);
+		if (values[0] instanceof Boolean) {
+			Boolean[] newValues = Arrays.stream(values)
+				.map(value -> (Boolean) value)
+				.map(value -> !value)
+				.toArray(Boolean[]::new);
+			expressions.change(e, newValues, ChangeMode.SET);
+			return;
+		}
+		
 		if (!flattening) {
 			executeLegacy(e);
 			return;
 		}
 		
 		// 1.13 and newer: use BlockData
-		for (Block b : blocks.getArray(e)) {
+		for (Block b : (Block[]) expressions.getArray(e)) {
```

Comment:
You're calling getArray twice on the same expression during one execution, you shouldn't do that